packages/react/src/utils/hasResetKeysChanged.spec.ts

```diff
@@ -5,49 +5,53 @@ const reference1 = { test: 0 }
+    it('if the two arrays have same lengths but at least one primitive element is different in each index of arrays.', () => {
+      const array1 = [primitive, primitive + 1]
+      const array2 = [primitive, primitive]
 
-    expect(hasResetKeysChanged(array1, array2)).toBe(false)
-  })
+      expect(hasResetKeysChanged(array1, array2)).toBe(true)
+    })
 
-  it('should return true if the two arrays have same lengths but at least one primitive element is different in each index of arrays.', () => {
-    const array1 = [primitive, primitive + 1]
-    const array2 = [primitive, primitive]
+    it('if the two arrays have same lengths and have all same primitive elements but order is different', () => {
+      const array1 = [primitive, primitive + 1]
+      const array2 = [primitive + 1, primitive]
 
-    expect(hasResetKeysChanged(array1, array2)).toBe(true)
-  })
+      expect(hasResetKeysChanged(array1, array2)).toBe(true)
+    })
 
-  it('should return true if the two arrays have same lengths and have all same primitive elements but order is different', () => {
-    const array1 = [primitive, primitive + 1]
-    const array2 = [primitive + 1, primitive]
+    it('if the two arrays have each references elements in index of array have different references', () => {
+      const array1 = [reference1, { test: 2 }]
+      const array2 = [reference1, { test: 2 }]
 
-    expect(hasResetKeysChanged(array1, array2)).toBe(true)
+      expect(hasResetKeysChanged(array1, array2)).toBe(true)
+    })
   })
 
-  it('should return true when two arrays have each references elements in index of array have different references', () => {
-    const array1 = [reference1, { test: 2 }]
-    const array2 = [reference1, { test: 2 }]
+  describe('should return false', () => {
+    it('if the two arrays have same lengths and same primitive element in each index of arrays.', () => {
+      const array1 = Array.from({ length: 1 }).map((_, index) => primitive + index)
+      const array2 = Array.from({ length: 1 }).map((_, index) => primitive + index)
 
-    expect(hasResetKeysChanged(array1, array2)).toBe(true)
-  })
+      expect(hasResetKeysChanged(array1, array2)).toBe(false)
+    })
 
-  it('should return false when two arrays have each references elements in index of array have same references', () => {
-    const array1 = [reference1, reference2]
-    const array2 = [reference1, reference2]
+    it('if the two arrays have each references elements in index of array have same references', () => {
+      const array1 = [reference1, reference2]
+      const array2 = [reference1, reference2]
 
-    expect(hasResetKeysChanged(array1, array2)).toBe(false)
-  })
+      expect(hasResetKeysChanged(array1, array2)).toBe(false)
+    })
 
-  it('should return false when no arrays as parameters. because of default value', () => {
-    expect(hasResetKeysChanged()).toBe(false)
+    it('if no arrays as parameters. because of default value', () => {
```

Comment:
https://daily-dev-tips.com/posts/jest-test-vs-it/

In my opinion, we need to use `it` correctly for better readability 
See post about `test` vs `it` please

Reply:
@manudeli
 Thank you for the great feedback. After considering it, I decided to revert the descriptions and separate the tests into describe blocks based on the return values.
  I'll look forward to your feedback on the current changes.

Reply:
Cool! Thanks for your improving us